e2e/e2e_test.go

```diff
@@ -459,3 +460,217 @@ func TestQuery(t *testing.T) {
 
 	// TODO: add tests for group by and subject
 }
+
+func TestCredit(t *testing.T) {
+	client := initClient(t)
+	customer := "customer-1"
+	productId := "credit_test_product_id"
+
+	// Reproducible random data
+	faker := gofakeit.New(8675309)
+	var events []cloudevents.Event
+
+	newEvent := func(timestamp string, model string) cloudevents.Event {
+		ts, err := time.Parse(time.RFC3339, timestamp)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ev := cloudevents.New()
+		ev.SetID(faker.UUID())
+		ev.SetSource("credit-test")
+		ev.SetType("credit_event")
+		ev.SetTime(ts)
+		ev.SetSubject("customer-1")
+		_ = ev.SetData("application/json", map[string]string{
+			"model": model,
+		})
+
+		return ev
+	}
+
+	// First event
+	{
+		events = append(events, newEvent("2024-01-01T00:01:00Z", "gpt-4"))
+	}
+
+	// Irrilevant event
+	{
+		events = append(events, newEvent("2024-01-01T00:01:00Z", "gpt-3"))
+	}
+
+	// Ingore events
+	{
+		resp, err := client.IngestEventBatchWithResponse(context.Background(), events)
+		require.NoError(t, err)
+		require.Equal(t, http.StatusNoContent, resp.StatusCode())
+	}
+
+	// Wait for events to be processed
+	time.Sleep(10 * time.Second)
```

Comment:
I think youshould implement a call that checks if the events are processed in the specific amount of time.

It not just makes integration tests faster, but also checks if the prerequisites for the next test cases have been fulfilled, thus helps with troubleshooting (e.g. it clearly shows if we have an issue with event processing or ledger).

Reply:
This pattern was already established. I opened a an issue to improve it: https://github.com/openmeterio/openmeter/issues/828